src/openforms/submissions/tests/test_start_submission.py

```diff
@@ -203,11 +199,20 @@ def test_start_submission_bad_form_url(self):
 
     @patch("openforms.logging.logevent._create_log")
     def test_start_submission_with_prefill(self, mock_logevent):
-        FormVariableFactory.create(
+        # we are creating a new step below-no need for two steps
+        self.form.formstep_set.get().delete()
+        FormStepFactory.create(
             form=self.form,
-            form_definition=self.step.form_definition,
-            prefill_plugin="demo",
-            prefill_attribute="random_string",
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "textfield",
+                        "key": "test-key",
+                        "label": "Test label",
+                        "prefill": {"plugin": "demo", "attribute": "random_string"},
+                    }
+                ]
+            },
```

Comment:
does this result in the same variable being created?

Reply:
I realized that in the way we test things here (getting the submission variables by `SubmissionValueVariable.objects.filter(submission=submission)` )gives us one submission variable when the form variables are two. I would do it with  `submission.load_submission_value_variables_state()` tbh which gives 2. I am a bit confused here..

Reply:
I traced back why the additional variable was created and patched that behaviour, so the test changes are reverted too :)